tests/test_dict.py

```diff
@@ -96,27 +96,31 @@ def test_setdefault_with_values(self):
     def test_setdefault_with_intervals(self):
         d = P.IntervalDict([(P.closed(0, 2), 0)])
         t = d.setdefault(P.closed(-2, -1), -1)
-        assert t.items() == [(P.closed(-2, -1), -1)]
-        assert d.items() == [(P.closed(-2, -1), -1), (P.closed(0, 2), 0)]
+        assert t.as_dict() == {P.closed(-2, -1): -1}
+        assert d.as_dict() == {P.closed(-2, -1): -1, P.closed(0, 2): 0}
 
         d = P.IntervalDict([(P.closed(0, 2), 0)])
         t = d.setdefault(P.closed(-1, 1), 2)
-        assert t.items() == [(P.closedopen(-1, 0), 2), (P.closed(0, 1), 0)]
-        assert d.items() == [(P.closedopen(-1, 0), 2), (P.closed(0, 2), 0)]
+        assert t.as_dict() == {P.closedopen(-1, 0): 2, P.closed(0, 1): 0}
+        assert d.as_dict() == {P.closedopen(-1, 0): 2, P.closed(0, 2): 0}
 
     def test_iterators(self):
         d = P.IntervalDict([(P.closedopen(0, 1), 0), (P.closedopen(1, 3), 1), (P.singleton(3), 2)])
 
-        assert d.keys() == [P.closedopen(0, 1), P.closedopen(1, 3), P.singleton(3)]
+        assert set(d.keys()) == set([P.closedopen(0, 1), P.closedopen(1, 3), P.singleton(3)])
         assert d.domain() == P.closed(0, 3)
-        assert d.values() == [0, 1, 2]
-        assert d.items() == list(zip(d.keys(), d.values()))
-        assert list(d) == d.keys()
+        assert set(d.values()) == set([0, 1, 2])
+        assert set(d.items()) == set([
+            (P.closedopen(0, 1), 0),
+            (P.closedopen(1, 3), 1),
+            (P.singleton(3), 2),
+        ])
+        assert set(d) == set(d.keys())
```

Comment:
I would also test that the returned view objects remain up-to-date with `d` when the latter changes after allocating the views.

Reply:
I added a test specifically for this, thanks!

Reply:
You’re welcome ;)